### PR TITLE
Add `flake8-rst-docstrings` and fix those broken in odoo core

### PR DESCRIPTION
**Pull request by xmo-odoo**
Lots of docstrings are wrong in the details with invalid rST, documenting parameters which don't exist (anymore), *not* documenting parameters which do, fucking up info fields, specifying types incorrectly, ...

This PR is a stab at fixing some of those issues and trying to make things last a bit, by enabling a checker to try and prevent broken docstrings from being committed. Note that the checker is currently configured to only check the core (odoo and the `base` module): this is already a fair amount of changes, so to avoid Extreme Conflicts I think it'd be a good idea to get that merged, then start expanding the scope of the checker.

### CODE REVIEW

**Comment by xmo-odoo** [comment]
So far this mostly fixes what pycharm can see, which is generally discrepancies between formal parameters and info fields but also regularly plain incorrectly-used info fields:

* `:param name type:` (other way around)
* C++/Java style generics
* `:type:` or `:returns:` with the typespec included in the tag

One persistent problem is that [pycharm doesn't support `kwargs` being documented individually via `params`](https://youtrack.jetbrains.com/issue/PY-50756).

Anyway after fixing most of those, trying to plug the addons into autodoc (via sphinx-apidoc + some trickery) yields a lot more issues:

* since Python 3.5, method overrides [automatically inherit the parent's docstring](https://docs.python.org/3/library/inspect.html?highlight=inspect#inspect.getdoc) which causes issues when the source docstring when e.g. it contains named targets (also leads to duplication of common ORM methods which is annoying)
* there's a bunch of long-deprecated epydoc-style info fields
* lots of docstrings are straight up not valid rST with random indentation and formatting, incorrect / incomplete cross-references, ...

**Comment by xmo-odoo** [comment]
NOTE: would probably be a good idea to test https://github.com/peterjc/flake8-rst-docstrings and add it into the CI pipeline (alongside flake8). Currently we generally don't want to impose docstrings (so the pydocstyle stuff) but the docstrings we have should be at least somewhat valid and not complete garbage.

**Comment by xmo-odoo** [comment]
@d-fence added config to [the existing `setup.cfg`](https://github.com/odoo/odoo/pull/74604/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52) in order to correctly configure the flake, but not sure how to make it run on runbot.

**Comment by xmo-odoo** [comment]
@robodoo override=ci/security

false positives from docstrings being updated

**Comment by xmo-odoo** [comment]
@AntoineVDV didn't you express interest in improving (and maintaining) docstrings so they could be more reliably integrated in the docs? This might be of interest to you.

File: odoo/models.py [hunk 1]
```diff
@@ -6742,18 +6763,27 @@ def _transient_vacuum(self):
...
         This unlinks old records from the transient model tables whenever the
-        "_transient_max_count" or "_max_age" conditions (if any) are reached.
-        Actual cleaning will happen only once every "_transient_check_time" calls.
-        This means this method can be called frequently called (e.g. whenever
-        a new record is created).
+        :attr:`_transient_max_count` or :attr:`_transient_max_hours` conditions
+        (if any) are reached.
+
+        Actual cleaning will happen only once every 5 minutes. This means this
+        method can be called frequently called (e.g. whenever a new record is
```
Comment (AntoineVDV) [inline odoo/models.py:6770]:
```suggestion
        method can be called frequently (e.g. whenever a new record is
```

**Comment by xmo-odoo** [comment]
@robodoo r+

**Pull request closed without merging**